src/game.cpp

```diff
@@ -9909,12 +9906,12 @@ bool game::handle_liquid(item &liquid, bool from_ground, bool infinite, item *so
 
     if( (liquid.type->id == "gasoline" || liquid.type->id == "diesel") &&
          vehicle_near() && query_yn(_("Refill vehicle?")) ) {
-        int vx = u.posx(), vy = u.posy();
+        tripoint vp;
```

Comment:
This needs to be initialized with `u.pos()`, otherwise the z-component is 0 and one can not refill a vehicle in an underground gas station. Tested with a debug spawned gas station at z=-1.